### PR TITLE
Refactor dashboard tab install with new DashboardTabModule

### PR DESCRIPTION
**Pull request by adrw**
Split out from https://github.com/cashapp/misk/pull/2800, the new DashboardTabModule provides an encapsulated way to install dashboard tabs. In future PRs it will be extended to provide support for installing IFrame and Hotwire tabs instead of only Misk-Web.

### CODE REVIEW

File: misk/src/main/kotlin/misk/web/resources/StaticResourceEntry.kt [hunk 1]
```diff
@@ -10,8 +10,8 @@ import misk.web.dashboard.ValidWebEntry
  * multibind<StaticResourceEntry>().toInstance(StaticResourceEntry(...))
  * ```
  */
-class StaticResourceEntry(
-  url_path_prefix: String = "/",
+data class StaticResourceEntry(
```
Comment (adrw) [inline misk/src/main/kotlin/misk/web/resources/StaticResourceEntry.kt:13]: Moving these to `data class` make for easier debugging given built-in `toString`, `equals`...etc

File: misk-admin/src/main/kotlin/misk/web/dashboard/AdminDashboardAccess.kt [hunk 2]
```diff
@@ -1,5 +1,16 @@
 package misk.web.dashboard
 
+/**
+ * Bind to set access for the Misk Admin Dashboard.
```
Comment (adrw) [inline misk-admin/src/main/kotlin/misk/web/dashboard/AdminDashboardAccess.kt:4]: Added missing docs

File: misk-admin/src/main/kotlin/misk/web/dashboard/DashboardTabModule.kt [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+     * @param developmentWebProxyUrl: url to resolve resource requests from in development ie. Webpack Dev Server
+     * @param resourcePathPrefix: path prefix used for background network requests to get resources
+     *   (HTML, CSS...) for the tab from a resource provider (classpath, filesystem, web proxy...).
+     * @param classpathResourcePathPrefix: location to find resources within classpath
+     * @param iframePath: complete path including file and extension if necessary which is set as the
+     *   iframe src attribute in the generated HTML.
+     * @param name: tab name which appears in the dashboard menu, usually titlecase
+     * @param category: menu category which the tab appears under
+     */
+    inline fun <reified DA : Annotation, reified AA : Annotation> createMiskWeb(
```
Comment (keeferrourke) [inline misk-admin/src/main/kotlin/misk/web/dashboard/DashboardTabModule.kt:30]: I don't love these docs

Comment (adrw) [reply misk-admin/src/main/kotlin/misk/web/dashboard/DashboardTabModule.kt:24]: I took another stab at it

**Pull request merged**